src/keypair.rs

```diff
@@ -0,0 +1,69 @@
+use algebra::{ProjectiveCurve, UniformRand};
+
+use crate::domain::*;
+
+use crate::pubkey::*;
+
+pub type SecKey = PallasScalar;
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct Keypair {
+    pub sec_key: SecKey,
+    pub pub_key: PubKey,
+}
+
+impl Keypair {
+    pub fn rand() -> Self {
+        let sec_key: PallasScalar = PallasScalar::rand(&mut rand_core::OsRng);
+        let pub_key: PallasPoint = PallasPoint::prime_subgroup_generator().mul(sec_key).into_affine();
+        return Keypair { sec_key: sec_key, pub_key: pub_key};
+    }
+
+    pub fn from_hex(sec_key_hex: &str) -> Result<Self, &'static str> {
+        let sec_key = PallasScalar::from_hex(sec_key_hex).or_else(
+            |_| Err("Invalid secret key hex")
+        )?;
+        let pub_key: PallasPoint = PallasPoint::prime_subgroup_generator().mul(sec_key).into_affine();
+        return Ok(Keypair { sec_key: sec_key, pub_key: pub_key});
+    }
+
+    pub fn address(self) -> String {
+        return self.pub_key.to_address();
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn from_hex() {
+        assert_eq!(Keypair::from_hex(""), Err("Invalid secret key hex"));
+        assert_eq!(Keypair::from_hex("1428fadcf0c02396e620f14f176fddb5d769b7de2027469d027a80142ef8f07"), Err("Invalid secret key hex"));
+        assert_eq!(Keypair::from_hex("0f5314f176fddb5d769b7de2027469d027ad428fadcf0c02396e6280142efb7d8"), Err("Invalid secret key hex"));
+        assert_eq!(Keypair::from_hex("g64244176fddb5d769b7de2027469d027ad428fadcf0c02396e6280142efb7d8"), Err("Invalid secret key hex"));
+        assert_eq!(Keypair::from_hex("dd4244176fddb5d769b7de2027469d027ad428fadcc0c02396e6280142efb718"), Err("Invalid secret key hex"));
+
+        let kp = Keypair::from_hex("164244176fddb5d769b7de2027469d027ad428fadcc0c02396e6280142efb718").expect(
+            "failed to decode keypair secret key");
+
+        println!("address = {}", kp.pub_key.to_address());
+    }
+
+    #[test]
+    fn address() {
```

Comment:
maybe a more descriptive name?

Reply:
I don't know how else to call it... it's a mina address.  maybe `mina_address`, but perhaps that's too much?

Reply:
I've been giving long sentence name to my tests, I'm not sure if there's a best practice but it's nice to see what fails when a test fails :o like `testing_if_address_deserialize_etc`.

Reply:
Ah, yes. good point. It comes out as
`test keypair::tests::address ... ok`

I'll update to something longer.

Reply:
Done